torchgeo/datasets/mapinwild.py

```diff
@@ -0,0 +1,479 @@
+            if (
+                download
+                and modal not in os.listdir(self.root)
+                and len(self.modality_urls[modal]) == 2
+            ):
+                self._merge_parts(modal)
+
+        # Masks will be loaded seperately in the :meth:`__getitem__`
+        if "mask" in self.modality:
+            self.modality.remove("mask")
+
+        # Split IDs has been downloaded and not needed in the list
+        if "split_ids" in self.modality:
+            self.modality.remove("split_ids")
+
+        if os.path.exists(os.path.join(self.root, "split_IDs.csv")):
+            split_dataframe = pd.read_csv(os.path.join(self.root, "split_IDs.csv"))
+            self.ids = split_dataframe[split].dropna().values.tolist()
+            self.ids = [int(i) for i in self.ids]
+
+    def __getitem__(self, index: int) -> dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        list_modals = []
+        id = self.ids[index]
+
+        mask = self._load_raster(id, "mask")
+        mask[mask != 0] = 1
+
+        for mode in self.modality:
+            mode = mode.upper() if mode in ["esa_wc", "viirs"] else mode
+            data = self._load_raster(id, mode)
+            list_modals.append(data)
+
+        image = torch.cat(list_modals, dim=0)
+
+        if self.transforms is not None:
+            sample_trans: dict[str, Tensor] = {"image": image, "mask": mask}
+
+            transformed = self.transforms(sample_trans)
+
+            image = rearrange(transformed["image"], "h w c -> c h w")
+            mask = rearrange(transformed["mask"], "h w c -> c h w")
+
+        sample: dict[str, Tensor] = {"image": image, "mask": mask}
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.ids)
+
+    def _load_raster(self, filename: int, source: str) -> Tensor:
+        """Load a single raster image or target.
+
+        Args:
+            filename: name of the file to load
+            source: the filename of the modality
+
+        Returns:
+            the raster image or target
+        """
+        with rasterio.open(os.path.join(self.root, source, f"{filename}.tif")) as f:
+            array = f.read()
+            if array.dtype == np.uint16:
+                array = array.astype(np.int32)
+            tensor = torch.from_numpy(array)
+            return tensor
+
+    def _verify(self, url: str, md5: Optional[str] = None) -> None:
+        """Verify the integrity of the dataset.
+
+        Args:
+            url: url to the file
+            md5: md5 of the file to be verified
+
+        Raises:
+            RuntimeError: if dataset is not found
+        """
+        modality_folder_name = os.path.split(url)[-1]
```

Comment:
URLs are not really "paths". On Windows, the path separator is \, not /. It would be better to split by / regardless of OS.